### PR TITLE
complete hm2

### PR DESCRIPTION
**Pull request by asyalu**

### CODE REVIEW

File: src/app/shared/components/info/info.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  templateUrl: './info.component.html',
+  styleUrls: ['./info.component.scss'],
+})
+export class InfoComponent implements OnInit {
+  @Input() title: string = 'title';
+  @Input() text: string = 'text';
+  @Input() buttonTitle: string = '';
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (MishkanJigan) [inline src/app/shared/components/info/info.component.ts:13]: I think it is possible to remove methods that are not used.

File: src/app/shared/components/header/header.component.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent implements OnInit {
+  userName: string = 'Tom';
+  buttonTitle: string = 'Logout';
+  constructor() {}
```
Comment (MishkanJigan) [inline src/app/shared/components/header/header.component.ts]: I think it is possible to remove constructor() that are not used.

File: src/app/shared/components/header/header.component.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<header class="header">
+  <img class="logo" src="src/assets/training.svg" />
```
Comment (MishkanJigan) [inline src/app/shared/components/header/header.component.html]: Try to use this one to resolve problem with image - ``.

File: src/app/shared/components/header/header.component.ts (hunk 2 above)
Comment (MishkanJigan) [reply src/app/shared/components/header/header.component.ts]: Review this point in all components.

File: src/app/shared/components/info/info.component.ts (hunk 1 above)
Comment (MishkanJigan) [reply src/app/shared/components/info/info.component.ts:13]: Review this point in all components.

File: src/app/features/course/course.component.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    <h3>{{ course.title }}</h3>
+    <div>{{ course.description }}</div>
+  </div>
+  <div>
+    <p>Authors: {{ course.authors }}</p>
+    <p>Duration: {{ course.duration }}</p>
+    <p>Created: {{ course.created }}</p>
+    <div class="controls">
+      <app-button [title]="buttonTitle" (click)="showButtons()"></app-button>
+      <app-button *ngIf="buttonsVisible" [icon]="editButton"></app-button>
```
Comment (MishkanJigan) [inline src/app/features/course/course.component.html:12]: Why hide these buttons?

File: src/app/features/course/course.component.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  <div>
+    <h3>{{ course.title }}</h3>
+    <div>{{ course.description }}</div>
+  </div>
+  <div>
+    <p>Authors: {{ course.authors }}</p>
+    <p>Duration: {{ course.duration }}</p>
+    <p>Created: {{ course.created }}</p>
+    <div class="controls">
+      <app-button [title]="buttonTitle" (click)="showButtons()"></app-button>
```
Comment (MishkanJigan) [inline src/app/features/course/course.component.html:11]: Try doing the same functionality, but for a modal window (a simple div with styles), not for other buttons.

File: src/app/shared/components/button/button.component.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss'],
+})
+export class ButtonComponent implements OnInit {
+  @Input() title?: string = '';
+  @Input() icon?: IconDefinition;
+
```
Comment (MishkanJigan) [inline src/app/shared/components/button/button.component.ts:13]: Try to add a button click handler here that will `.emit()` a event to the top component with `@Output()`.
And use this output in other components where you work with `app-button`.

```
@Output() clickButton = new EventEmitter();

  onClick() {
    this.clickButton.emit();
  }
```

File: src/app/features/course/course.component.html (hunk 4 above)
Comment (MishkanJigan) [reply src/app/features/course/course.component.html:12]: It is better to add some click handler here, which will simply display which button was clicked in the console, or activate the modal window, make it visible.

File: src/app/features/course/course.component.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+})
+export class CourseComponent implements OnInit {
+  @Input() course: ICourse = courses[0];
+  @Output() buttonsVisible: boolean = false;
+  buttonTitle: string = 'Show course';
+  editButton: IconDefinition = faPencil;
+  deleteButton: IconDefinition = faTrash;
+  constructor() {}
+
+  showButtons() {
```
Comment (MishkanJigan) [inline src/app/features/course/course.component.ts]: If you trying to use typescript, add return type to all functions, as example in this function we have type - void.

File: src/app/features/course/course.component.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<div class="course">
+  <div>
+    <h3>{{ course.title }}</h3>
+    <div>{{ course.description }}</div>
+  </div>
+  <div>
+    <p>Authors: {{ course.authors }}</p>
+    <p>Duration: {{ course.duration }}</p>
```
Comment (MishkanJigan) [inline src/app/features/course/course.component.html:8]: Try to use date pype  - `| date: 'dd.MM.yyyy'` to change the format of the date.

File: src/app/shared/components/info/info.component.ts (hunk 1 above)
Comment (asyalu) [reply src/app/shared/components/info/info.component.ts:13]: image

File: src/app/shared/components/header/header.component.html (hunk 3 above)
Comment (asyalu) [reply src/app/shared/components/header/header.component.html]: yes it helped

File: src/app/features/course/course.component.html (hunk 4 above)
Comment (asyalu) [reply src/app/features/course/course.component.html:12]: I will add this in the next MR

File: src/app/features/course/course.component.html (hunk 5 above)
Comment (asyalu) [reply src/app/features/course/course.component.html:11]: I will add this in the next MR

File: src/app/shared/components/button/button.component.ts (hunk 6 above)
Comment (asyalu) [reply src/app/shared/components/button/button.component.ts:13]: done

File: src/app/features/course/course.component.ts (hunk 7 above)
Comment (asyalu) [reply src/app/features/course/course.component.ts]: done

File: src/app/features/course/course.component.html (hunk 8 above)
Comment (asyalu) [reply src/app/features/course/course.component.html:8]: executed in next MP

**Pull request merged**